### PR TITLE
feat:created new layout for multimeter

### LINKED ISSUE #1110: Use colours in multimeter instead of line for classifications
**Actual Behaviour**

Currently lines are used for classification in multimeter

**Expected Behaviour**

Colours should be used instead of lines

**Steps to reproduce it**

Go to multimeter layout and resolve it

**LogCat for the issue**

Not applicable

**Screenshots of the issue**

Not applicable

**Would you like to work on the issue?**

Yes

### PR DESCRIPTION
**Pull request by abhinavraj23**
Fixes #1110 #1163 #1083 

**Changes**: 
Added colors in multimeter

**Screenshot/s for the changes**:

 

**Checklist**: [Please tick following check boxes with `[x]` if the respective task is completed]
- [x] I have used resources from `strings.xml`, `dimens.xml` and `colors.xml` without hard-coding them
- [x] No modifications done at the end of resource files `strings.xml`, `dimens.xml` or `colors.xml`
- [x] I have reformatted code in every file included in this PR [<kbd>CTRL</kbd>+<kbd>ALT</kbd>+<kbd>L</kbd>]
- [x] My code does not contain any extra lines or extra spaces
- [x] I have requested reviews from other members

**APK for testing**: 
[NewUI.apk.zip](https://github.com/fossasia/pslab-android/files/2173651/NewUI.apk.zip)

### CODE REVIEW

**Comment by abhinavraj23** [comment]
@CloudyPadmal @Avjeet @harsh-2711 @Vikum94 please suggest changes and how we can add colors in a better way

**Comment by harsh-2711** [comment]
@abhinavraj23 These type of colors don't look good. Please use some suitable colors and also try to use as less colors as possible. Group some of the sections with same color as the one on original multimeter

**Comment by Vikum94** [comment]
Try to find if there are standard colors or designs that goes with multi-meters.

File: app/src/main/res/layout/activity_multimeter.xml [hunk 1]
```diff
@@ -618,10 +337,183 @@
...
+        app:layout_constraintTop_toTopOf="@+id/knob_line_center2"
         app:layout_constraintVertical_bias="0.489" />
 
+    <TextView
+        android:layout_width="wrap_content"
+        android:layout_height="wrap_content"
+        android:layout_gravity="center"
+        android:gravity="center"
+        android:text="@string/ch1"
+        android:textColor="#f9710a"
```
Comment (Vikum94) [inline app/src/main/res/layout/activity_multimeter.xml]: Add all the color strings to resource files

**Comment by CloudyPadmal** [comment]
Also remove the bars as discussed a few times in weekly meetings.

**Comment by Avjeet** [comment]
@abhinavraj23 These colors are not looking good and also try to customize the knob.

**Comment by abhinavraj23** [comment]
@CloudyPadmal @Avjeet @harsh-2711 I have made the changes and have used only the primary colors and black, I have also removed the lines.

**Comment by abhinavraj23** [comment]
multi_scr
This is the screenshot

**Comment by akarshan96** [comment]
@abhinavraj23 here are some points I think should be implemented in th UI.
- Using rounded borders will make the layout look better. Try using rounded_custom_border.xml around the knob. 

-  Most of the multimeters have grey screens like we have in calculators. I think display area in the UI should be grey too and you can use digital font to display the outputs (refer [this](https://stackoverflow.com/questions/9276615/is-it-possible-to-show-digital-texts-in-textview-in-android)).

- Align the components.
screen shot 2018-06-23 at 2 04 25 pm

- You can also do something that has been done in Wave Generator.
screen shot 2018-06-23 at 7 03 25 pm

@CloudyPadmal @Avjeet @Vikum94 @harsh-2711 add your views to these points too.

**Comment by Avjeet** [comment]
@abhinavraj23 I think the most important part of your UI is knob you should first customize that.

**Comment by CloudyPadmal** [comment]
Yes I  agree with the suggestions made by @akarshan96 and @Avjeet !

**Comment by Vikum94** [comment]
Check here how the different groupings are done around the knob
image

**Comment by abhinavraj23** [comment]
multimeter_newui
@CloudyPadmal @harsh-2711 @Vikum94 @Avjeet this is the new UI, please comment and let me if it is fine or what changes are required.

**Comment by CloudyPadmal** [comment]
Yes @abhinavraj23 this looks nice!

**Comment by Avjeet** [comment]
@abhinavraj23 This is not center aligned remove the left padding.

**Comment by abhinavraj23** [comment]
@CloudyPadmal @harsh-2711 @Vikum94 @CloudyPadmal Please review it now I have done all the necessary changes

**Comment by abhinavraj23** [comment]
[multimeter_newUIapk.zip](https://github.com/fossasia/pslab-android/files/2135714/multimeter_newUIapk.zip) this is the latest apk

**Comment by Avjeet** [comment]
@abhinavraj23 Can u first plz change the name of PR to something relevant as you are changing the layout as well and not the colors only.

File: app/src/main/res/layout/activity_multimeter.xml [hunk 2]
```diff
@@ -419,46 +208,44 @@
...
-        android:id="@+id/kno_line"
+        android:id="@+id/knob_line_up"
         android:layout_width="match_parent"
-        android:layout_height="@dimen/multimeter_knoblines_width"
-        android:layout_marginEnd="@dimen/multimeter_square_margin_right"
-        android:layout_marginLeft="@dimen/multimeter_square_margin_left"
-        android:layout_marginRight="@dimen/multimeter_square_margin_right"
-        android:layout_marginStart="@dimen/multimeter_square_margin_left"
-        android:layout_marginTop="@dimen/multimeter_square_horizontal_line_margin_top"
+        android:layout_height="1dp"
```
Comment (Avjeet) [inline app/src/main/res/layout/activity_multimeter.xml]: you can replace these hard coded dimen

File: app/src/main/res/layout/activity_multimeter.xml [hunk 3]
```diff
@@ -273,69 +16,116 @@
...
+        android:background="@drawable/rectangle_border"
+        app:layout_constraintBottom_toTopOf="@+id/textView11"
+        app:layout_constraintEnd_toStartOf="@+id/textView6"
+        app:layout_constraintHorizontal_bias="1.0"
+        app:layout_constraintStart_toStartOf="parent"
+        app:layout_constraintTop_toBottomOf="@+id/resistance"
+        app:layout_constraintVertical_bias="1.0" />
+
     <View
         android:id="@+id/view4"
```
Comment (Avjeet) [inline app/src/main/res/layout/activity_multimeter.xml]: Can you use some relevant IDs here

**Comment by abhinavraj23** [comment]
@Avjeet Changes done please review now. Also @harsh-2711 @CloudyPadmal @Vikum94 please review

**Comment by abhinavraj23** [comment]
@Avjeet Changes done :+1: , @CloudyPadmal @harsh-2711 @Vikum94 Please review

**Comment by Avjeet** [comment]
@abhinavraj23 Resolve the conflicts

**Comment by CloudyPadmal** [comment]
@abhinavraj23 please resolve the conflicts and upload the latest APK. In the PR body APK is not available. Broken link

**Comment by abhinavraj23** [comment]
@CloudyPadmal Changes done 
[NewUI.apk.zip](https://github.com/fossasia/pslab-android/files/2151565/NewUI.apk.zip)
Here is the new APK
@harsh-2711 @Vikum94 @Avjeet Please review

**Comment by harsh-2711** [comment]
@abhinavraj23 Please center align the text as done by @Avjeet for main headings. Also as shown in the screenshot, the AN8 and CH3 pins have more padding than CH1 and CAP pins. Please make it uniform.
screenshot_20180701-001855

**Comment by CloudyPadmal** [comment]
Also @abhinavraj23 please update the PR body screenshot as well when you make changes

**Comment by Avjeet** [comment]
@abhinavraj23 are you implementing knob in this PR?

**Comment by abhinavraj23** [comment]
screenshot_20180705-185520
@CloudyPadmal @harsh-2711 @Vikum94 @Avjeet Have a look at the new implementation and comment

**Comment by abhinavraj23** [comment]
@CloudyPadmal Please add the work in progress title. The new implementation will be ready by tomorrow as I have to also change the java codes correspondingly :+1:

**Comment by harsh-2711** [comment]
@abhinavraj23 Don't use units around the knob as it's confusing for a user to know what's Vc, CP, etc. Do use the functionality name. For example frequency, capacitance, etc. And for units, you can mention them in the description itself or behind the main text around the knob.

**Comment by abhinavraj23** [comment]
screenshot_20180706-165713
@CloudyPadmal @harsh-2711 @Vikum94 @Avjeet Please review this. @CloudyPadmal I have changed the knob as per suggestion, I have also used the card views for radio- buttons. Please suggest what else can be done :+1:

**Comment by abhinavraj23** [comment]
Also we can give the abbreviations in the guide itself :+1:

**Comment by abhinavraj23** [comment]
@CloudyPadmal Let this PR be just for the UI. I will implement the backend part in a new PR, is this fine ?

**Comment by CloudyPadmal** [comment]
@abhinavraj23 is that screen an actual implementation from Android Studio or any kind of an image editing? Because there is a different font.
Also by card view I meant the whole description thing can be put inside the card

**Comment by abhinavraj23** [comment]
@CloudyPadmal That is the actual implementation from  android studio. I have deliberately kept the font like that as looked good, should I change it? . I will put the whole description in card-view :+1:

**Comment by abhinavraj23** [comment]
@CloudyPadmal I have added the whole description in card-view. Shall I implement a slider for card-views or shall I normally change the text and pins inside. I think for the starting let us keep the normal text changing and then afterwards we can implement a slider :+1:

**Comment by abhinavraj23** [comment]
@harsh-2711 , @CloudyPadmal said i for the previous knob, I think this one looks good also in an actual multimeter there are standard abbreviations for capacitance, resistance voltage and frequency which I will make the change but there is no standard abbreviation for count pulse, so can you suggest what can be better than CP, as we cannot write full words(it doesn't fit).

**Comment by harsh-2711** [comment]
@abhinavraj23 You can use count/min or count/sec.

File: app/src/main/res/layout/activity_multimeter.xml [hunk 4]
```diff
@@ -545,83 +157,352 @@
...
+        app:layout_constraintCircleRadius="@dimen/multimeter_knobcircle_radius_1" />
+
+    <TextView
+        android:layout_width="wrap_content"
+        android:layout_height="wrap_content"
+        android:layout_gravity="center"
+        android:gravity="center"
+        android:text="@string/resistance_abbreviation"
+        android:textColor="@color/black"
+        android:textSize="18sp"
```
Comment (Vikum94) [inline app/src/main/res/layout/activity_multimeter.xml]: This is a large file so you have missed few places where you can use resource files.

File: app/src/main/res/layout/activity_multimeter.xml [hunk 5]
```diff
@@ -545,83 +157,352 @@
...
+        app:layout_constraintCircle="@id/knobs"
+        app:layout_constraintCircleAngle="240"
+        app:layout_constraintCircleRadius="@dimen/multimeter_knobcircle_radius_1" />
+
+    <ImageView
+        android:layout_width="25dp"
+        android:layout_height="20dp"
```
Comment (Vikum94) [inline app/src/main/res/layout/activity_multimeter.xml]: Here and following image view

**Comment by abhinavraj23** [comment]
@Vikum94 @CloudyPadmal @harsh-2711 @Avjeet Now I have made all the changes in this PR, please review it, also @CloudyPadmal please remove the WIP tag. This is the latest image

**Pull request closed without merging**